lib/src/main/scala/spinal/lib/com/uart/UartCtrl.scala

```diff
@@ -191,10 +191,10 @@ object UartCtrl {
 }
 
 case class UartCtrlInitConfig(
-  baudrate : Int = 0,
-  dataLength : Int = 0,
-  parity : UartParityType.E = null,
-  stop : UartStopType.E = null
+  var baudrate : Int = 0,
```

Comment:
Should these really be vars?
The user can never make those vals again so that they stay consistent, but if you want/need to be able to change them you can always have a `var config: UartCtrlInitConfig` and reassign to that from `config.copy(...)` (as with pretty much all other configs in lib)

Reply:
Overall, i'm more and more "pissed" by the all functional paradigm / val ^^
config.copy is kinda too dogmatic i think. While mutability is just easy ?
Often it work just great to have a default config and mutate it a bit.

> (as with pretty much all other configs in lib)

Maybe that is wrong aswell

Reply:
Not sure - If you pass one config object into multiple constructors (forgetting to copy them) and one ends up changing it, it's suddenly modified for all places. Kind of the curse of Java reference semantics.

Reply:
> Not sure - If you pass one config object into multiple constructors (forgetting to copy them) and one ends up changing it, it's suddenly modified for all places. Kind of the curse of Java reference semantics.

Yes, right.

On the other hand, imutability prevent "natural" patterns as : 

```scala
      val uart = new TilelinkUartFiber()
      uart.config.initConfig.baudrate = 9600
```

Forcing you to get everything ahead of time through parameters, preventing preset default values.

In other words, imutability feel a bit like (in another context) "everything private in class, use only getters/setters"